static/app/components/modals/sudoModal.tsx

```diff
@@ -44,76 +53,90 @@ type Props = WithRouterProps &
     retryRequest?: () => Promise<any>;
   };
 
-type State = {
-  authenticators: Array<Authenticator>;
-  busy: boolean;
-  error: boolean;
-  errorType: string;
-  showAccessForms: boolean;
-  superuserAccessCategory: string;
-  superuserReason: string;
-};
-
-class SudoModal extends Component<Props, State> {
-  static defaultProps: DefaultProps = {
-    closeButton: true,
-  };
-
-  state: State = {
-    authenticators: [],
-    busy: false,
+function SudoModal({
+  closeModal,
+  isSuperuser,
+  location,
+  needsReload,
+  router,
+  retryRequest,
+  api,
+  Header,
+  Body,
+  closeButton,
+}: Props) {
+  const [state, setState] = useState<State>({
+    authenticators: [] as Authenticator[],
     error: false,
     errorType: '',
     showAccessForms: true,
     superuserAccessCategory: '',
     superuserReason: '',
-  };
+  });
+
+  const {
+    authenticators,
+    error,
+    errorType,
+    showAccessForms,
+    superuserAccessCategory,
+    superuserReason,
+  } = state;
+
+  useEffect(() => {
+    const getAuthenticators = async () => {
+      try {
+        const fetchedAuthenticators = await api.requestPromise('/authenticators/');
+        setState(prevState => ({
+          ...prevState,
+          authenticators: fetchedAuthenticators ?? [],
+        }));
+      } catch {
+        // ignore errors
+      }
+    };
 
-  componentDidMount() {
-    this.getAuthenticators();
-  }
+    getAuthenticators();
+  }, [api]);
```

Comment:
Is this `api` dependency necessary? I tried removing it, but the the test `can delete an org with sudo flow` fails with:
```
Error: No mocked response found for request: GET /authenticators/
```
even though it's being mocked in the `beforeEach` and the actual test passes

Reply:
yes because we are using it in the useEffect